### DISCUSSION TITLE
Get all incoming nodes ( direct / indirect )

### DISCUSSION THREAD

**Discussion by navaniharsh31** [xyflow/xyflow#2890] [Q&A]
I have a graph with plenty of nodes, few might be connected nodeId to nodeId and a few are connected through custom handles inside a node to nodeId.

How do i find the full chain of connected nodes to specific node id?

**Comment by moklick** [comment]
You can use the helper functions `getIncomers` and `getOutgoers` https://reactflow.dev/docs/api/graph-util-functions/#getoutgoers
